lms/services/blackboard_api/service.py

```diff
@@ -53,12 +53,38 @@ def list_files(self, course_id):
         :raise ProxyAPIAccessTokenError: if we don't have a Blackboard API
             access token for the current user
         """
-        return self._http_service.get(
+        files = self._http_service.get(
             self._api_url(f"courses/uuid:{course_id}/resources"),
             oauth=True,
             schema=BlackboardListFilesSchema,
         ).validated_data
 
+        return [
+            {
+                "id": self._blackboard_document_url(file_["id"]),
+                "display_name": file_["name"],
+                "updated_at": file_["modified"],
+            }
+            for file_ in files
+        ]
+
+    def public_url(self, course_id, file_id):
+        files = self._http_service.get(
+            self._api_url(f"/courses/uuid:{course_id}/resources"),
+            oauth=True,
+            schema=BlackboardListFilesSchema,
+        ).validated_data
+
+        for file_ in files:
+            if self._blackboard_document_url(file_["id"]) == file_id:
+                return file_["downloadUrl"]
+
+        assert False
+
+    @staticmethod
+    def _blackboard_document_url(file_id):
+        return f"blackboard://content-resource/{file_id}/"
```

Comment:
I think this could have a docstring, I'd try to look for this schema on blackboard's docs first time seeing this.

Also `content-resource` seems a bit vague, I'll explicitly include `file` somewhere so they are more self explanation if seen without context.

Reply:
Added a docstring